community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/StorePropertyPayloadCursor.java

```diff
@@ -87,6 +87,7 @@
 
     private final long[] data = new long[MAX_NUMBER_OF_PAYLOAD_LONG_ARRAY];
     private int position = INITIAL_POSITION;
+    private boolean exhausted;
```

Comment:
After this cursor was exhausted all subsequent `#next()` calls threw exceptions. This flag guards this condition. Test that reproduces the issue is here: https://github.com/neo4j/neo4j/pull/7060/files#diff-c2f8ebee67771699e30477c42da1c63dR165